src/blockchain/node/wallet/subchain/SubchainStateData.cpp

```diff
@@ -1104,32 +1092,23 @@ auto SubchainStateData::scan(
     try {
         using namespace std::literals;
         const auto procedure = rescan ? "rescan"sv : "scan"sv;
-        const auto& log = log_;
-        const auto& name = name_;
-        const auto& node = node_;
         const auto& type = filter_type_;
-        const auto& headers = node.HeaderOracle();
-        const auto& filters = node.FilterOracleInternal();
+        const auto& headers = node_.HeaderOracle();
+        const auto& filters = node_.FilterOracleInternal();
         const auto start = Clock::now();
         const auto startHeight = highestTested.first + 1;
         auto atLeastOnce = std::atomic_bool{false};
         auto highestClean = std::optional<block::Position>{std::nullopt};
         auto resultMap = [&] {
```

Comment:
there is still this lambda not changed but whole function must be refactored which is not simple. I created jira ticket for it